src/code/z_room.c

```diff
@@ -1,94 +1,579 @@
+
+            // Project the entry position, to get the depth it is at.
+            pos.x = roomShapeCullableEntry->boundsSphereCenter.x;
+            pos.y = roomShapeCullableEntry->boundsSphereCenter.y;
+            pos.z = roomShapeCullableEntry->boundsSphereCenter.z;
+            SkinMatrix_Vec3fMtxFMultXYZ(&play->viewProjectionMtxF, &pos, &projectedPos);
+            // SkinMatrix_Vec3fMtxFMultXYZW(&play->viewProjectionMtxF, &pos, &projectedPos, &projectedW);
+
+            projectedPos.z *= var_fa1;
+
+            var_fv1 = ABS_ALT(roomShapeCullableEntry->boundsSphereRadius);
+
+            // If the entry bounding sphere isn't fully before the rendered depth range
+            if (-var_fv1 < projectedPos.z) {
+
+                // Compute the depth of the nearest point in the entry's bounding sphere
+                entryBoundsNearZ = projectedPos.z - var_fv1;
+
+                // If the entry bounding sphere isn't fully beyond the rendered depth range
+                if (entryBoundsNearZ < play->lightCtx.unkC) {
+
+                    // This entry will be rendered
+                    insert->entry = roomShapeCullableEntry;
+
+                    if (roomShapeCullableEntry->boundsSphereRadius < 0) {
+                        insert->boundsNearZ = FLT_MAX;
+                    } else {
+                        insert->boundsNearZ = entryBoundsNearZ;
+                    }
+
+                    // Insert into the linked list, ordered by ascending depth of the nearest point in the bounding
+                    // sphere
+                    iter = head;
+                    if (iter == NULL) {
+                        head = tail = insert;
+                        insert->prev = insert->next = NULL;
+                    } else {
+                        do {
+                            if (insert->boundsNearZ < iter->boundsNearZ) {
+                                break;
+                            }
+                            iter = iter->next;
+                        } while (iter != NULL);
+
+                        if (iter == NULL) {
+                            insert->prev = tail;
+                            insert->next = NULL;
+                            tail->next = insert;
+                            tail = insert;
+                        } else {
+                            insert->prev = iter->prev;
+                            if (insert->prev == NULL) {
+                                head = insert;
+                            } else {
+                                insert->prev->next = insert;
+                            }
+                            iter->prev = insert;
+                            insert->next = iter;
+                        }
+                    }
+
+                    insert++;
+                }
+            }
+        }
+
+        // if this is real then I might not be
+        R_ROOM_CULL_NUM_ENTRIES = roomShape->numEntries & 0xFFFF & 0xFFFF & 0xFFFF;
```

Comment:
rofl

Reply:
copy-pasta from OoT, I guess I can remove it. Interestingly seems to be needed in MM as well (or at least this fake trick also works in MM)